### ISSUE TITLE
Why does it fetch everything even when using --since-tag?

### ISSUE THREAD

**Issue by LulzAugusto** [github-changelog-generator/github-changelog-generator#645]
Hey, I work on a kinda large project and I wanna integrate this tool with our release process. The problem is that it takes one run to reach the API limit and it seems that even using parameters like `since-tag` it still shows `fetching all the 800+ tags`. Is this something I can help with maybe or is there a reason for it to work like this? I really think we can make it more useful by focusing on increasing changes instead of always regenerating the whole file. 

Another question: what does the basefile option do exactly? Does it actually read the file so only the new tags will be added?

All in all, great work everyone, this tool generates the best output by far! 🎉

**Comment by LulzAugusto** [comment]
>Is this something I can help with maybe or is there a reason for it to work like this? I really think we can make it more useful by focusing on increasing changes instead of always regenerating the whole file.

Anyone? 🙏

**Comment by hunner** [comment]
tl;dr the github api only supports fetching all tags afaict.

---

The API method used to fetch tags is the Repository#tags method at https://github.com/github-changelog-generator/github-changelog-generator/blob/24a1b457877581182cf9f9a189f90ab33c73b472/lib/github_changelog_generator/octo_fetcher.rb#L111-L113

This API method does not take any filtering options that I am aware of, according to https://developer.github.com/v3/repos/#list-tags via https://octokit.github.io/octokit.rb/Octokit/Client/Repositories.html#tags-instance_method so it's an all-or-nothing operation.

The `tags` API doesn't even return the dates of the tags, so that has to be fetched in a separate operation: https://github.com/github-changelog-generator/github-changelog-generator/blob/24a1b457877581182cf9f9a189f90ab33c73b472/lib/github_changelog_generator/generator/generator_fetcher.rb#L18-L27 and since there is no guarantee that tags are fetched in date-order they must all be fetched.

Something similar would be how the API *does* allow filtering listing commits with a `since=<ISO8601 date>` parameter: https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository

I haven't looked into how this would affect timing, but there might be some perf gains in throwing away all tags older than the date of `since_tag` once they've been sorted by https://github.com/github-changelog-generator/github-changelog-generator/blob/24a1b457877581182cf9f9a189f90ab33c73b472/lib/github_changelog_generator/generator/generator_tags.rb#L53-L59
